### PR TITLE
fix: post url and post photo preview

### PR DESCRIPTION
**Pull request by LeifXu**
# Pull Request

## Related Issue

Closes FW-0000

## Checklist

-   [ ] I have tested these changes locally.
-   [ ] I have reviewed the code changes.
-   [ ] I have added unit tests if applicable.

### CODE REVIEW

**Review by nuanyang233 — DISMISSED** [review]

File: src/components/Posts/SinglePost.tsx [hunk 1]
```diff
@@ -51,14 +50,12 @@ export const SinglePost = memo<SinglePostProps>(function SinglePost({
 
     const pathname = usePathname();
 
-    const isPostPage = isRoutePathname(pathname, '/post/:detail', true);
-    const isProfilePage = isRoutePathname(pathname, '/profile/:id', true);
-    const isChannelPage = isRoutePathname(pathname, '/channel/:detail', true);
+    const isPostPage = isRoutePathname(pathname, '/post/:source/:id', true);
+    const isProfilePage = isRoutePathname(pathname, '/profile/:source/:id/:category', true);
+    const isChannelPage = isRoutePathname(pathname, '/channel/:detail/:type', true);
```
Comment (swkatmask) [inline src/components/Posts/SinglePost.tsx:55]: What about
```suggestion
    const isPostPage = isRoutePathname(pathname, '/post/:source/');
    const isProfilePage = isRoutePathname(pathname, '/profile/:source/');
    const isChannelPage = isRoutePathname(pathname, '/channel/:detail/');
```

**Review by swkatmask — APPROVED** [review]

File: src/helpers/parsePostUrl.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { isSocialSource } from '@/helpers/isSocialSource.js';
+import { resolveSourceFromUrl } from '@/helpers/resolveSource.js';
+
+export function parseOldPostUrl(url: URL) {
+    if (!url.pathname.startsWith('/post')) return null;
+    const sourceInUrl = url.searchParams.get('source') ?? '';
```
Comment (guanbinrui) [inline src/helpers/parsePostUrl.ts:6]:
```suggestion
    const sourceInUrl = url.searchParams.get('source') ?? '';
    if (!sourceInUrl) return null;
```

File: src/helpers/isSocialSource.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
-import { type SocialSource, type SocialSourceInURL, Source, SourceInURL } from '@/constants/enum.js';
+import { type SocialSource, type SocialSourceInURL, SourceInURL } from '@/constants/enum.js';
 import { SORTED_SOCIAL_SOURCES } from '@/constants/index.js';
 import { resolveSourceInURL } from '@/helpers/resolveSourceInURL.js';
 
-export function isSocialSource(source?: Source): source is SocialSource {
+export function isSocialSource(source?: string): source is SocialSource {
```
Comment (guanbinrui) [inline src/helpers/isSocialSource.ts:5]: Use force type casting inplace.

File: src/helpers/parsePostUrl.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+export function parseOldPostUrl(url: URL) {
+    if (!url.pathname.startsWith('/post')) return null;
+    const source = resolveSourceFromUrlNoFallback(url.searchParams.get('source'));
+    if (!source || !isSocialSource(source)) return null;
+    const [, , id, ...end] = url.pathname.split('/');
+    if (end.join('/') !== '') return null;
+    if (isSocialSource(id)) return null;
+    if (!id) return null;
+    return { source, id };
+}
```
Comment (guanbinrui) [inline src/helpers/parsePostUrl.ts]:
```suggestion
export function parseOldPostUrl(url: URL) {
    if (!url.pathname.startsWith('/post')) return null;
    const source = resolveSourceFromUrlNoFallback(url.searchParams.get('source'));
    if (!source || !isSocialSource(source)) return null;
    const [, , id] = url.pathname.split('/');
    if (!id) return null;
    return { source, id };
}
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**